src/nvim/terminal.c

```diff
@@ -205,6 +202,8 @@ void terminal_teardown(void)
 
 Terminal *terminal_open(TerminalOptions opts)
 {
+  // Will this be a true color terminal?
+  bool true_color = os_getenv("NVIM_TUI_ENABLE_TRUE_COLOR") != NULL;
```

Comment:
Will that tell me if NVIM_TUI_ENABLE_TRUE_COLOR is set or not? I mean if it only tells me if the terminal is capable of true color, then it is not enough as it will break index colors if the user sets g:terminal_color colors and has not enabled NVIM_TUI_ENABLE_TRUE_COLOR.